apps/api/jwt_auth/handlers.py

```diff
@@ -0,0 +1,77 @@
+from datetime import datetime
+import logging
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+
+import jwt
+from rest_framework_jwt.settings import api_settings
+
+from apps.api.models import APIKey, SYMMETRIC_JWT_TYPE
+
+log = logging.getLogger('z.jwt')
+
+
+def get_jwt_key(key):
+    return APIKey.objects.get(key=key, type=SYMMETRIC_JWT_TYPE)
+
+
+def jwt_payload_handler(user, issuer):
+    issued_at = datetime.utcnow()
+    return {
+        # The JWT issuer must match the 'key' field of APIKey
+        'iss': issuer,
+        'iat': issued_at,
+        'exp': issued_at + api_settings.JWT_EXPIRATION_DELTA,
+    }
+
+
+def jwt_encode_handler(payload, secret):
+    token = jwt.encode(payload, secret, api_settings.JWT_ALGORITHM)
+    return token.decode('utf-8')
+
+
+def jwt_decode_handler(token, get_api_key=get_jwt_key):
+    token_data = jwt.decode(token, verify=False)
```

Comment:
This is the first pass at decoding where it just literally decodes the base64 json data so it can look at the issuer ID. It uses that ID to look up the secret. In the next pass (several lines down) it decodes again with `verify=True` which checks the signature with the secret. Having to do two passes like this is one of the annoying things about JWTs. JSON-LD is a little bit nicer but I don't think it has as many developer libraries yet.